File: MLR_DummyVariable.py
# -*- coding: utf-8 -*-
"""
Created on 08/15/2019
Assignment on Multiple Linear Regression - Dummy Variables
Purpose: Create Dummy variables on categorical variable
Dataset: iris [builtin r]
3 categories
@author: Haresh
"""
import pandas as pd

iris1 = pd.read_csv('D:\\Documents - SSD\\iris.csv')
iris1.Species.value_counts()

#Option 1: insert new columns manually using map function of pandas using dicts

iris1["Species_setosa"] = iris1.Species.map({'setosa':1,'versicolor':0,'virginica':0})
iris1["Species_versicolor"] = iris1.Species.map({'setosa':0,'versicolor':1,'virginica':0})
iris1.head()

#Option 2: insert new columns using get_dummies in pandas, drops original column and first dummy variable, adding others.

iris1 = pd.read_csv('D:\\Documents - SSD\\iris.csv')
iris1.Species.value_counts()

iris1 = pd.get_dummies(iris1, columns=['Species'], prefix_sep='_', drop_first=True)
iris1.head()

#Option 3: using get_dummies and keeping the original column

iris1 = pd.read_csv('D:\\Documents - SSD\\iris.csv')
iris1.Species.value_counts()

iris1 = pd.concat([iris1, pd.get_dummies(iris1.Species, prefix='Species').iloc[:,:-1]], axis=1)
iris1.head()

File: MLR_DummyVariable.R
# Assignment on Multiple Linear Regression - Dummy Variables
# Created on 08/14/2019
# Purpose: Create Dummy variables on categorical variable
# Dataset: iris [builtin r]
# 3 categories

iris1 <- iris # Copies to an internal dataset iris1 to work on the dataset
summary(iris1)
attach(iris1) #Petal.Length, Petal.Width, Sepal.Length, Sepal.Width, Species
levels(Species) # display how many categories this variable is split into
table(is.na(Species)) # making sure there are no NA values in the categories


#Create new variables under iris1. We need 2 as there are 3 categories of Species
#variable if Xa == 1, its setosa and Xb == 1, its versicolor, if Xa == Xb == 0, its virginica
#Option 1 using for and if
for (i in 1:length(Species))
  {
  iris1$Xa[i] <- ifelse(Species[i] == "setosa", 1, 0)
  iris1$Xb[i] <- ifelse(Species[i] == "versicolor", 1, 0)
  #print(Species[i])
  }
str(iris1)

#Option 2 using cbind and dummies
#install.packages("dummies")
iris1 <- iris # Copies to an internal dataset iris1 to work on the dataset
library(dummies)
iris1 <- cbind(iris1, dummy(Species, verbose = TRUE))
str(iris1)

## To download the builtin datasets for external usage

write.csv(iris,file="iris.csv", row.names = FALSE)
getwd()
